src/rust/engine/concrete_time/src/lib.rs

```diff
@@ -102,7 +102,14 @@ impl TimeSpan {
     end: &std::time::SystemTime,
   ) -> TimeSpan {
     let start = Self::since_epoch(start);
-    let duration = Self::since_epoch(end) - start;
+    let end = Self::since_epoch(end);
+    let duration = match end.checked_sub(start) {
+      Some(d) => d,
+      None => {
+        log::warn!("Invalid TimeSpan - start: {:?}, end: {:?}", start, end);
```

Comment:
If this is happening fairly often, we're not going to want to be logging warnings, probably. But it would also be good to track down which codepath is creating inverted ranges like this. So maybe drop to debug...?

It should be easy to detect that this is happening due to zero-length workunits.